### PR TITLE
fix(html): Add advice on object element with data attr instead of param element

### LINKED ISSUE #31523: `<param>` attribute deprecated - missing sources
### MDN URL

https://developer.mozilla.org/en-US/docs/Web/HTML/Element/param

### What specific section or headline is this issue about?

deprecated: `<param>`: The Object Parameter element

### What information was incorrect, unhelpful, or incomplete?

It is said that this element is deprecated, contrary to multiple sources. It was deprived of a very specific behavior, but there are still use cases like providing dynamic parameters to an SVG image.

* https://chromestatus.com/feature/6283184588193792 <-- `<param>` was not removed, just a particular behaviour/attribute is no longer a part of its implementation

### What did you expect to see?

Either sources for deprecation statement (along with W3C recommendation or a doc) and a verified link to one of browsers' code that would suggest `<param>` element removal.

### Do you have any supporting links, references, or citations?

* https://chromestatus.com/feature/6283184588193792
* https://www.w3.org/TR/SVGParamPrimer/
* https://www.w3.org/TR/2012/WD-html-markup-20121025/param.html - pretty old
* latest spec: https://html.spec.whatwg.org/ (no word about <param> element being deprecated or obsolete)

### Do you have anything more you want to share?

_No response_

### MDN metadata

<details>
<summary>Page report details</summary>

* Folder: `en-us/web/html/element/param`
* MDN URL: https://developer.mozilla.org/en-US/docs/Web/HTML/Element/param
* GitHub URL: https://github.com/mdn/content/blob/main/files/en-us/web/html/element/param/index.md
* Last commit: https://github.com/mdn/content/commit/942a529383ee7ee3996fb234187641c08935f3ff
* Document last modified: 2023-04-13T06:13:15.000Z

</details>

### PR DESCRIPTION
**Pull request by bsmth**
### Description

The feature is deprecated and https://github.com/mdn/content/issues/31523 asks for rationale for this status.

As linked in the issue, https://github.com/mdn/content/pull/15225 is the PR that introduces this status change in the docs.

__Additions:__

* Note that points to an alternative 

### Related issues and pull requests

Fixes https://github.com/mdn/content/issues/31523

### CODE REVIEW

File: files/en-us/web/html/element/param/index.md [hunk 1]
```diff
@@ -11,6 +11,9 @@ browser-compat: html.elements.param
 
 The **`<param>`** [HTML](/en-US/docs/Web/HTML) element defines parameters for an {{HTMLElement("object")}} element.
 
+> **Note:**
+> It's recommended to use the {{HTMLElement("object")}} element with a [data attribute](/en-US/docs/Web/HTML/Element/object#data) to set the URL of an external resource instead.
```
Comment (skyclouds2001) [inline files/en-us/web/html/element/param/index.md:15]:
```suggestion
> It's recommended to use the {{HTMLElement("object")}} element with a [`data` attribute](/en-US/docs/Web/HTML/Element/object#data) to set the URL of an external resource instead.
```

Comment (bsmth) [reply files/en-us/web/html/element/param/index.md:15]: Nice, thanks 👍🏻

File: files/en-us/web/html/element/param/index.md [hunk 2]
```diff
@@ -11,6 +11,9 @@ browser-compat: html.elements.param
 
 The **`<param>`** [HTML](/en-US/docs/Web/HTML) element defines parameters for an {{HTMLElement("object")}} element.
 
+> **Note:**
+> It's recommended to use the {{HTMLElement("object")}} element with a [`data` attribute](/en-US/docs/Web/HTML/Element/object#data) to set the URL of an external resource instead.
```
Comment (dipikabh) [inline files/en-us/web/html/element/param/index.md:15]: I guess we can be more direct:
```suggestion
> **Note:**
> Use the {{HTMLElement("object")}} element with a [`data`](/en-US/docs/Web/HTML/Element/object#data) attribute to set the URL of an external resource.
```

OR

this version to remove the straggling "instead":

```suggestion
> **Note:**
> As a preferred alternative, use the {{HTMLElement("object")}} element with a [`data` attribute](/en-US/docs/Web/HTML/Element/object#data) to set the URL of an external resource.
```

**Review by dipikabh — APPROVED** [review]
Thanks for adding this note, @bsmth.
Minor suggestions but leaving a +1 so you can merge when ready.

**Comment by bsmth** [comment]
Thanks, all, merging shortly 👍🏻

**Pull request merged**